pkg/collect/collector.go

```diff
@@ -67,7 +67,7 @@ func GetCollector(collector *troubleshootv1beta2.Collect, bundlePath string, nam
 	case collector.ConfigMap != nil:
 		return &CollectConfigMap{collector.ConfigMap, bundlePath, namespace, clientConfig, client, ctx, RBACErrors}, true
 	case collector.Logs != nil:
-		return &CollectLogs{collector.Logs, bundlePath, namespace, clientConfig, client, ctx, sinceTime, RBACErrors}, true
+		return &CollectLogs{collector.Logs, bundlePath, namespace, client, ctx, sinceTime, RBACErrors}, true
```

Comment:
All collectors are passing both.
Which I agree that is strange why I would pass the client and the config?

Besides I go along with the change I am favor to have standards always.
So, what about it be done in a follow up and for all case(s) instead of change only this one?
On top of that, would have some reason for we are getting a new client for each case? 

Anyway, I think would be very beneficial a follow up here where we would chose by default one of the both. I mean, we pass the config and get a new client always or we use the same client for all case scenarios. So that we can discuss if has a reason to create a new client.  

See https://pkg.go.dev/k8s.io/client-go/kubernetes#NewForConfig so we might need to create a new Client for each case because of rate-limiter .

Reply:
I'm for creating fewer clients (one at best) that have the required rate limiting configurations (QPS, Burst, HTTPClient etc) rather than creating many clients. Trying to circumvent rate limiting by having many clients usually ends up failing in the long run cause if the API server is struggling, it will respond with errors.

Reply:
> Anyway, I think would be very beneficial a follow up here where we would chose by default one of the both. I mean, we pass the config and get a new client always or we use the same client for all case scenarios. So that we can discuss if has a reason to create a new client.

I agree. This needs to be looked at holistically in a separate issue/proposal

Reply:
We also need to remember that creating support bundles usually takes place in production systems. We'd rather get rate limited and handle it accordingly (retry, error out) rather than put more pressure on the systems.

Reply:
Hi @banjoh,

> I agree. This needs to be looked at holistically in a separate issue/proposal

I think we both are in the same page here that changes done here would better fit in a new PR. Right?
So, I'd say let's to discuss the motivations there. I am not sure If I can follow up your argumentations here and I think too much comments came make harder others follow up and help you out in the review.

So, could you revert this changes and push them in another pr?

Reply:
> So, could you revert the clientConfig changes and push it in another pr? WDYT?

It makes sense. Change done